Feature/tee (#2)
iExecBlockchainComputing/floor-price-checker: README.md

```

### Input

> This solution reads an input file. You can then choose between two formats.  

Reviewer: ```suggestion
This solution reads either an input file or a dataset. You can then choose between two formats.  
```

1) Either you enter directly the Ethereum address of your wallet containing the nfts
```
<input.txt>
0x0123456789012345678901234567890123456789
```

> 2) Or, you can scpeficy the Collections (and numbers of items) you want to track :  

Reviewer 2: ```suggestion
2) Or, you can specify the Collections (and numbers of items) you want to track :  
```

> Here is an exemple below for the case => You own 2 Nfts of the first collection, 1 of the 2nd and 3 of the 3rd. (Don't forget to add "```Collections:```" on the first line of your file)

Reviewer 2: ```suggestion
Here is an example below for the case => You own 2 NTFs of the first collection, 1 of the 2nd and 3 of the 3rd. (Don't forget to add "`Collections:`" on the first line of your file)
```

```

### Input
This solution reads a .json input file following this format :
```
{

>     "ownerAddress": 0x_owner_address,

Reviewer 2: Shouldn't "0x_owner_address" be enclosed in double quotes, as all strings should be in JSON?

Author: yes definitely 👍🏼

    "collections": [
        {
            "collectionId": "collection_id",
            "count":  number_of_asset_owned
        },
        ...
    ]
}
```

> If you fill the `ownerAddress`, the dapp will directly ask Opensea for the collections and nft that you own, so you doesn't have to manually fill the `"collections"` part :

Reviewer 2: ```suggestion
If you fill the `ownerAddress`, the dapp will directly ask Opensea for the collections and nft that you own, so you don't have to manually fill the `collections` part :
```

```
{
    "ownerAddress": "0x01234567891012345678910123456789101234567891",

>     "collections": []

Reviewer 2: Is `collections` mandatory in that case?

Author: In that case it's not indeed

}
```

> **Or**, If you want to inspect specific collections (and not a 0x address), you can fill the `"collections"` part with the Opensea collection id (or slug) and the number of asset that you own from that collection :

Reviewer 2: ```suggestion
**Or**, If you want to inspect specific collections (and not a 0x address), you can fill the `collections` part with the Opensea collection id (or slug) and the number of assets that you own from that collection :
```

> ```{

Reviewer 2: There should be a blank line after those threes backquotes

>     "ownerAddress": "",

Reviewer 2: Is `ownerAddress` mandatory in that case?

Author: Ok unexpectedly it works so it's not mandatory, thanks !

Reviewer 2: In Go, `string` seems to be a [basic type](https://go.dev/tour/basics/11). Its default value/zero-value is `""`, not `nil`. A non-set `string` var has then the value `""`, so you can compare it safely to the empty string.

### Deploy
To deploy your app, follow the instructions on the IExec Documentation : https://docs.iex.ec/for-developers/your-first-app

Then, you can run your dApp with the ```iexec app run``` command (you can add as much parameters and options as you want, follow the SDK and CLI documentation to do so) :  
```
iexec app run --watch
```

### Confidential Computing and TEE
In order to benefit from the computation confidentiality offered by Trusted Execution Environnements, we first need to sconify our dApp.  

> To do that, just run the ```./sconify.sh``` script.  

Reviewer 2: ```suggestion
To do this, just run the following:  
```

```
./sconify.sh
```
It will build a sconified docker image of the app, that you can deploy the same way as a Standard dApp (like you did before following the iExec documentation).  
The code will now run inside a private enclave.  

You just have to add the ```--tag tee``` option in your run command :
```
iexec app run --watch --tag tee
```

But moreover, you can also add layer of confidentiality by protecting your input and output data.

### Datasets

> Following this documentation https://docs.iex.ec/for-developers/confidential-computing/sgx-encrypted-dataset, you will be able to encrypt your input file and then give your "secret" (encryption key) to the SMS (Secret Management Service). Like this, no one (except you) will be able to read what your input data was.

Reviewer 2: ```suggestion
Following [this documentation](https://docs.iex.ec/for-developers/confidential-computing/sgx-encrypted-dataset), you will be able to encrypt your input file and then give your "secret" (encryption key) to the SMS (Secret Management Service). Like this, no one (except you) will be able to read what your input data is.
```

Author: Thanks for the tips 👍🏼

### End to End Encryption

> Finally, in order to achieve End to End encryption, you can encrypt your result following this documentation https://docs.iex.ec/for-developers/confidential-computing/end-to-end-encryption

Reviewer 2: ```suggestion
Finally, in order to achieve End-to-End encryption, you can encrypt your result following [this documentation](https://docs.iex.ec/for-developers/confidential-computing/end-to-end-encryption).
```
